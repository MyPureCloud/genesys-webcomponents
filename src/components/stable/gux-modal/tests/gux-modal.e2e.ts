import { newE2EPage } from '@stencil/core/testing';

describe('gux-modal', () => {
  describe('#render', () => {
    [
      {
        description: 'should render small modal',
        html: `
          <gux-modal lang="en" size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render medium modal',
        html: `
          <gux-modal lang="en" size="medium">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render large modal',
        html: `
          <gux-modal lang="en" size="large">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without a title',
        html: `
          <gux-modal lang="en" size="large">
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without buttons',
        html: `
          <gux-modal lang="en" size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just left align buttons',
        html: `
          <gux-modal lang="en" size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just right align buttons',
        html: `
          <gux-modal lang="en" size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render small modal by default',
        html: `
          <gux-modal lang="en">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="left-align-buttons">
                <gux-button title="Cancel">Cancel</gux-button>
            </div>
            <div slot="right-align-buttons">
              <gux-button title='Button' accent='primary'>Accept</gux-button>
            </div>
          </gux-modal>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newE2EPage({ html });
        const element = await page.find('gux-modal');

        expect(element.outerHTML).toMatchSnapshot();

        element.setAttribute('hidden', true);
        await page.waitForChanges();

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <gux-modal lang="en" size="small">
          <div slot="title">Modal Title</div>
          <div slot="content">This contains the modal content.</div>
          <div slot="left-align-buttons">
              <gux-button title="Cancel">Cancel</gux-button>
          </div>
          <div slot="right-align-buttons">
            <gux-button title='Button' accent='primary'>Accept</gux-button>
          </div>
        </gux-modal>
      `;
      const page = await newE2EPage({ html });
      const element = await page.find('gux-modal');
      const dismissButton = await page.find('.gux-dismiss-button');
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-modal')).not.toBeNull();

      await dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-modal')).toBeNull();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <gux-modal lang="en" size="small">
          <div slot="title">Modal Title</div>
          <div slot="content">This contains the modal content.</div>
          <div slot="left-align-buttons">
              <gux-button title="Cancel">Cancel</gux-button>
          </div>
          <div slot="right-align-buttons">
            <gux-button title='Button' accent='primary'>Accept</gux-button>
          </div>
        </gux-modal>
      `;
      const page = await newE2EPage({ html });
      const element = await page.find('gux-modal');
      const dismissButton = await page.find('.gux-dismiss-button');
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      await page.evaluate(() => {
        document.addEventListener('guxdismiss', event => {
          event.preventDefault();
        });
      });

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-modal')).not.toBeNull();

      await dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-modal')).not.toBeNull();
    });
  });
});
