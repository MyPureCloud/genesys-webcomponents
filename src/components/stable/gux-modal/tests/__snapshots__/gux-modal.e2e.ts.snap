// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-modal #render should render large modal 1`] = `"<gux-modal lang=\\"en\\" size=\\"large\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container large\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render large modal 2`] = `"<gux-modal lang=\\"en\\" size=\\"large\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container large\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render medium modal 1`] = `"<gux-modal lang=\\"en\\" size=\\"medium\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container medium\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render medium modal 2`] = `"<gux-modal lang=\\"en\\" size=\\"medium\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container medium\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal with just left align buttons 1`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal with just left align buttons 2`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal with just right align buttons 1`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal with just right align buttons 2`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal without a title 1`] = `"<gux-modal lang=\\"en\\" size=\\"large\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container large\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal without a title 2`] = `"<gux-modal lang=\\"en\\" size=\\"large\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container large\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal without buttons 1`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content no-buttons\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render modal without buttons 2`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content no-buttons\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render small modal 1`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render small modal 2`] = `"<gux-modal lang=\\"en\\" size=\\"small\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render small modal by default 1`] = `"<gux-modal lang=\\"en\\" class=\\"hydrated\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;

exports[`gux-modal #render should render small modal by default 2`] = `"<gux-modal lang=\\"en\\" class=\\"hydrated\\" hidden=\\"true\\"><!----> <div class=\\"modal\\"><div class=\\"modal-container small\\"><button class=\\"dismiss-button\\" title=\\"Dismiss\\"><gux-icon screenreader-text=\\"Dismiss\\" icon-name=\\"ic-close\\" class=\\"hydrated\\"><div class=\\"container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"> <path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path> </svg></div></gux-icon></button><h1 class=\\"modal-header\\"><div slot=\\"title\\">Modal Title</div></h1><div class=\\"modal-content\\"><p></p><div slot=\\"content\\">This contains the modal content.</div><p></p></div><div class=\\"button-footer\\"><div class=\\"left-align-buttons\\"><div slot=\\"left-align-buttons\\"> <gux-button title=\\"Cancel\\" class=\\"hydrated\\"><!----><button title=\\"Cancel\\" class=\\"gux-secondary\\">Cancel</button></gux-button> </div></div><div class=\\"right-align-buttons\\"><div slot=\\"right-align-buttons\\"> <gux-button title=\\"Button\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><button title=\\"Button\\" class=\\"gux-primary\\">Accept</button></gux-button> </div></div></div></div></div></gux-modal>"`;
